controllers/metric_controller.go

```diff
@@ -156,40 +162,34 @@ func (r *MetricReconciler) collectMetrics(ctx context.Context, t *redskyv1beta1.
 			continue
 		}
 
-		// Capture the metric
-		var captureError error
-		if target, err := r.target(ctx, t.Namespace, metrics[v.Name]); err != nil {
-			captureError = err
-		} else if value, stddev, err := metric.CaptureMetric(log, metrics[v.Name], t, target); err != nil {
-			if merr, ok := err.(*metric.CaptureError); ok && merr.RetryAfter > 0 {
-				// Do not count retries against the remaining attempts
-				return &ctrl.Result{RequeueAfter: merr.RetryAfter}, nil
-			}
-			captureError = err
-		} else {
-			v.AttemptsRemaining = 0
-			v.Value = strconv.FormatFloat(value, 'f', -1, 64)
-			if stddev != 0 {
-				v.Error = strconv.FormatFloat(stddev, 'f', -1, 64)
-			}
+		// Use a defaulted DEEP COPY of the metric so we can safely make changes
+		m := metricWithDefaults(t, metrics[v.Name])
+
+		// This is strictly for converted v1alpha1 experiments
+		if err := r.resolveLegacyURL(ctx, t, m); err != nil {
+			return r.collectionAttempt(ctx, log, t, v, probeTime, err)
 		}
```

Comment:
probably just me ( so maybe uber-nit?) , but I'd put this with `metricWithDefaults`

Reply:
The reason it's not is that resolution of the legacy URL requires state from the reconciler (the client), while the current defaults implementation is a global function. I'll adjust things by making defaults and legacy URL resolution more coupled.